sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java

```diff
@@ -0,0 +1,435 @@
+    }
+
+    Mono<Response<OciManifest>> downloadManifestWithResponse(String tagOrDigest, Context context) {
+        if (tagOrDigest == null) {
+            return monoError(logger, new NullPointerException("'tagOrDigest' can't be null."));
+        }
+
+        return this.registriesImpl.getManifestWithResponseAsync(repositoryName, tagOrDigest, UtilsImpl.OCI_MANIFEST_MEDIA_TYPE, context)
+            .flatMap(response -> {
+                String digest = response.getHeaders().getValue(UtilsImpl.DOCKER_DIGEST_HEADER_NAME);
+                ManifestWrapper wrapper = response.getValue();
+
+                // The service wants us to validate the digest here since a lot of customers forget to do it before consuming
+                // the contents returned by the service.
+                if (Objects.equals(digest, tagOrDigest) || Objects.equals(response.getValue().getTag(), tagOrDigest)) {
+                    OciManifest ociManifest = new OciManifest()
+                        .setAnnotations(wrapper.getAnnotations())
+                        .setConfig(wrapper.getConfig())
+                        .setLayers(wrapper.getLayers())
+                        .setSchemaVersion(wrapper.getSchemaVersion());
+
+                    Response<OciManifest> res = new ResponseBase<Void, OciManifest>(
+                        response.getRequest(),
+                        response.getStatusCode(),
+                        response.getHeaders(),
+                        ociManifest,
+                        null);
+
+                    return Mono.just(res);
+                } else {
+                    return monoError(logger, new ServiceResponseException("The digest in the response does not match the expected digest."));
+                }
+            }).onErrorMap(UtilsImpl::mapException);
+    }
+
+    /**
+     * Download the blob associated with the given digest.
+     *
+     * @param digest The digest for the given image layer.
+     * @return The image associated with the given digest.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code digest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DownloadBlobResult> downloadBlob(String digest) {
+        return this.downloadBlobWithResponse(digest).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Download the blob\layer associated with the given digest.
+     *
+     * @param digest The digest for the given image layer.
+     * @return The image associated with the given digest.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code digest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DownloadBlobResult>> downloadBlobWithResponse(String digest) {
```

Comment:
Same here and in other places - `tagOrDigest`

Reply:
As above.